rasa/server.py

```diff
@@ -1152,13 +1158,11 @@ async def evaluate_intents(
                     test_data, config_file, int(cross_validation_folds)
                 )
         else:
-            test_data = _test_data_file_from_payload(request, temporary_directory)
-            if cross_validation_folds:
-                raise ErrorResponse(
-                    HTTPStatus.BAD_REQUEST,
-                    "TestingError",
-                    "Cross-validation is only supported for YAML data.",
-                )
+            raise ErrorResponse(
+                HTTPStatus.BAD_REQUEST,
+                "TestingError",
+                "NLU evaluation is only supported for YAML data.",
```

Comment:
Why is JSON removed as option?

Reply:
I'm not sure why, but it was removed already previously - it just wasn't documented that it was. Even passing a JSON body, it would get interpreted as YAML and throw an error.

Reply:
That's strange, according to this [deprecation warning](https://rasa.com/docs/rasa/next/migration-guide/#nlu-json-format) in the migration guide, the JSON nlu format should only be removed in 4.0 🤔

Reply:
Ah right I don't mean it was removed everywhere, but it was removed for this endpoint specifically.

Reply:
If you think it should still be supported, then I'll revert this and fix the handling of JSON to work

Reply:
I think we should be consistent and support it here too, maybe worth adding a deprecation warning that this format will be removed from this endpoint in 4.0.

Reply:
have added a method for it.